src/store/Disks.cc

```diff
@@ -46,12 +53,20 @@ objectSizeForDirSelection(const StoreEntry &entry)
     return minSize;
 }
 
+static SwapDir &
+SwapDirByIndex(const int i)
+{
+    const auto sd = INDEXSD(i);
```

Comment:
I do not think the INDEXSD hack should be completely removed in this PR because, even after the additional polishing in Disks.cc, some of the remaining hack uses (outside of Disks.cc) will need non-trivial API refactoring that is best done in a dedicated PR or two. We should keep `INDEXSD` as a marker for those places that still need significant work.

However, there is probably no good justification for keeping `INDEXSD` uses in Disks.cc (and where the call can be replaced with a StoreEntry::disk() call). Modernizing those trivial callers can be brought into in this PR scope without creating review problems AFAICT. @eduard-bagdasaryan, please do that. Just do not make any new disk-by-id APIs public. Keeping those APIs confined to Disks.cc will help control proliferation of callers that iterate cache_dirs outside of Disks.cc.